cradle-admin-tool-http/scripts/update-pages-comment.sh

```diff
@@ -0,0 +1,360 @@
+  else
+    echo " ERROR: '${START_TIMESTAMP}' start timestamp is greater than '${END_TIMESTAMP}' end timestamp."
+    exit 4
+  fi
+}
+
+verify_comment() {
+  if [ -z "${COMMENT}" ]; then
+    echo " ERROR: '${COMMENT}' comment can't be empty"
+    exit 5
+  fi
+}
+
+verify_mode() {
+  if [[ "${MODE}" != "${MODE_APPEND}" && "${MODE}" != "${MODE_SET}" && "${MODE}" != "${MODE_RESET}" ]]; then
+    echo " ERROR: '${MODE}' mode isn't either of values ['${MODE_APPEND}','${MODE_SET}','${MODE_RESET}']"
+    exit 6
+  fi
+}
+
+escape_json() {
+  local text
+  local escaped_text
+  text="${1}"
+  escaped_text=$(echo -n "${text}" | jq -Rsa .)
+  echo -n "${escaped_text}"
+}
+
+escape_http_parameter() {
+  local text
+  local escaped_text
+  text="${1}"
+  escaped_text=$(echo -n "${text}" | jq -sRr @uri)
+  echo -n "${escaped_text}"
+}
+
+escape_http_parameter_test() {
+  echo "Test"
+  escape_http_parameter '["abc","def"]'
+
+  exit 0
+}
+
+prepare_comment() {
+  local indent
+  local current_comment
+  local text
+  local new_comment
+  local escaped_current_comment
+  local escaped_comment
+
+  current_comment="${1}"
+  text="${2}"
+  indent="${3}"
+
+  escaped_comment=$(escape_json "${text}")
+  echo "${indent}Append '${text}' text to '${current_comment}' current comment" >&2
+  if [[ "${current_comment}" == "${CRADLE_ADMIN_DEFAULT_COMMENT}" ]]; then
+    echo "${indent} INFO: '${current_comment}' current comment is default. '${CRADLE_ADMIN_DEFAULT_COMMENT}' will be removed" >&2
+    new_comment="[${escaped_comment}]"
+    echo "${new_comment}"
+    return
+  fi
+
+  if [ -z "${current_comment}" ]; then
+    echo "${indent} INFO: '${current_comment}' current comment is empty." >&2
+    new_comment="[${escaped_comment}]"
+    echo "${new_comment}"
+    return
+  fi
+
+  if echo "${current_comment}" | jq -e 'if type == "array" then true else false end' > /dev/null 2>&1; then
+    echo "${indent} INFO: '${current_comment}' is JSON string array" >&2
+    if [[ "${current_comment}" == *"${escaped_comment}"* ]]; then
+      echo "${indent} INFO: '${current_comment}' current comment already contains ${escaped_comment}" >&2
+      new_comment=''
+    else
+      echo "${indent} INFO: Append '${escaped_comment}' into '${current_comment}' JSON string array" >&2
+      new_comment=$(echo "${current_comment}" | jq -c ". + [${escaped_comment}]")
+    fi
+  else
+    escaped_current_comment=$(escape_json "${current_comment}")
+    if [[ "${escaped_current_comment}" == "${escaped_comment}" ]]; then
+      echo "${indent} INFO: '${escaped_current_comment}' and '${escaped_comment}' are equal" >&2
+      new_comment="[${escaped_current_comment}]"
+    else
+      echo "${indent} INFO: Wrap '${escaped_current_comment}' and '${escaped_comment}' into JSON string array" >&2
+      new_comment="[${escaped_current_comment},${escaped_comment}]"
+    fi
+  fi
+
+  echo "${new_comment}"
+}
+
+prepare_comment_test() {
```

Comment:
note: this is also a test function if I understand correctly. Maybe we can extract them into a separate script?

Reply:
I would like to leave *_test functions for future changes